lib/query/lib/prepareForDelivery.js

```diff
@@ -24,13 +26,31 @@ export default (node) => {
     storeOneResults(node, node.results);
 }
 
-export function applyPostFilter(node) {
-    if (node.props.$postFilter) {
-        node.results = sift(node.props.$postFilter, node.results);
+export function applyPostFilters(node) {
+    const postFilters = node.props.$postFilters || node.props.$postFilter;
+    if (postFilters) {
+        node.results = sift(postFilters, node.results);
     }
 
     node.collectionNodes.forEach(collectionNode => {
-        applyPostFilter(collectionNode);
+        applyPostFilters(collectionNode);
+    })
+}
+export function applyPostOptions(node) {
+    const options = node.props.$postOptions;
+    if (options) {
+        if(options.limit || options.skip){
+            const start = options.skip || 0, end = options.limit ? options.limit + start : -0;
```

Comment:
I thought -0 would work, as negative numbers represent offset from the end of the array. But I tested it now and it doesn't actually work!
Changing it to node.results.length